Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -198,18 +199,35 @@ private extension TabScrollingController {
         return y
     }
 
-    func animateToolbarsWithOffsets(animated animated: Bool, duration: NSTimeInterval, headerOffset: CGFloat,
+    func animateToolbarsWithOffsets(animated animated: Bool, duration: NSTimeInterval, headerOffset: CGFloat, scrollOffset: CGFloat,
         footerOffset: CGFloat, alpha: CGFloat, completion: ((finished: Bool) -> Void)?) {
 
+        // Due to the header not being placed on top of the scroll view, when it animates away, and the scroll offset is negative,
+        // its superview background will show through temporarily, leading to anpleasant visual effect. By inserted a subview of the
+        // same colour as the background, we can hide this effect for the duration of the animation.
+        var veil: UIView?
+        if let header = self.header, superview = header.superview where animated {
```

Comment:
When I disable the veil, I see a black background behind the URL bar before it animates down to match up with the scroll view. Is this the graphical oddity you saw? If so, it seems that the timing of the animation of the URL bar doesn't match up with the velocity of the scroll view being put back in place. It's easier to see when I turn on slow animations in the simulator. I assume that this worked before when the views were connected by a constraint but now it's set to an offset from the topLayoutGuide.

I also notice that sometimes theres a bit of white that appears above the black area as well. We explicitly make the web content _not_ clip to the bounds so it's entirely likely that the glitch you're seeing is the web content poking through as well.

![url_bar_jank](https://cloud.githubusercontent.com/assets/434322/16996135/e6550218-4e7d-11e6-8476-06d8baa30457.gif)

I wonder if we can use the velocity from the scroll view snapping back into place to determine the animation speed so that the URL bar animates in at the time without that delay. That might fix these issues you're seeing and avoid the need for a 'viel' subview to hide it.

Thoughts?